### ISSUE TITLE
Remote SSH - Resolver error: Error: Connecting with SSH timed out

### ISSUE THREAD

**Issue by jeff94040** [microsoft/vscode-remote-release#7423]
Within VS Code, I can remote into Linux hosts successfully, but cannot seem to remote into a Windows 7 host. I installed OpenSSH server on the remote Windows 7 machine per `https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH-Using-MSI`

I feel like I've tried every combination of remote.SSH settings, nothing seems to work. 

Log with `"remote.SSH.useLocalServer": true`

```
[18:55:29.341] Log Level: 2
[18:55:29.342] remote-ssh@0.90.1
[18:55:29.343] darwin x64
[18:55:29.348] SSH Resolver called for "ssh-remote+192.168.0.12", attempt 1
[18:55:29.349] "remote.SSH.useLocalServer": true
[18:55:29.349] "remote.SSH.path": undefined
[18:55:29.349] "remote.SSH.configFile": undefined
[18:55:29.349] "remote.SSH.useFlock": true
[18:55:29.349] "remote.SSH.lockfilesInTmp": false
[18:55:29.350] "remote.SSH.localServerDownload": auto
[18:55:29.350] "remote.SSH.remoteServerListenOnSocket": false
[18:55:29.350] "remote.SSH.showLoginTerminal": true
[18:55:29.350] "remote.SSH.defaultExtensions": []
[18:55:29.350] "remote.SSH.loglevel": 2
[18:55:29.350] "remote.SSH.maxReconnectionAttempts": 0
[18:55:29.350] "remote.SSH.enableDynamicForwarding": true
[18:55:29.350] "remote.SSH.enableRemoteCommand": false
[18:55:29.351] "remote.SSH.serverPickPortsFromRange": {}
[18:55:29.351] "remote.SSH.serverInstallPath": {}
[18:55:29.370] SSH Resolver called for host: 192.168.0.12
[18:55:29.370] Setting up SSH remote "192.168.0.12"
[18:55:29.377] Acquiring local install lock: /var/folders/qz/4vbkfwp5093b0dnq9bdhll140000gn/T/vscode-remote-ssh-8bbfac03-install.lock
[18:55:29.379] Looking for existing server data file at /Users/schmidt/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-8bbfac03-d045a5eda657f4d7b676dedbfa7aab8207f8a075-0.90.1/data.json
[18:55:29.379] Using commit id "d045a5eda657f4d7b676dedbfa7aab8207f8a075" and quality "stable" for server
[18:55:29.384] Install and start server if needed
[18:55:29.393] PATH: /opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/schmidt/.rvm/bin
[18:55:29.394] Checking ssh with "ssh -V"
[18:55:29.408] > OpenSSH_8.1p1, LibreSSL 2.7.3

[18:55:29.415] askpass server listening on /var/folders/qz/4vbkfwp5093b0dnq9bdhll140000gn/T/vscode-ssh-askpass-2ee054eede8b263e1882054b1e3b51b9c913c51a.sock
[18:55:29.416] Spawning local server with {"serverId":1,"ipcHandlePath":"/var/folders/qz/4vbkfwp5093b0dnq9bdhll140000gn/T/vscode-ssh-askpass-cfa3c3cc63edd3743dd44e4b1ddb47de06420eb1.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","55246","-o","ConnectTimeout=15","192.168.0.12","powershell"],"serverDataFolderName":".vscode-server","dataFilePath":"/Users/schmidt/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-8bbfac03-d045a5eda657f4d7b676dedbfa7aab8207f8a075-0.90.1/data.json"}
[18:55:29.416] Local server env: {"SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.JUy9ZTdz8N/Listeners","SHELL":"/bin/bash","DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/schmidt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.90.1/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"/Users/schmidt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.90.1/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/qz/4vbkfwp5093b0dnq9bdhll140000gn/T/vscode-ssh-askpass-2ee054eede8b263e1882054b1e3b51b9c913c51a.sock"}
[18:55:29.432] Spawned 1306
[18:55:29.642] > local-server-1> Spawned ssh, pid=1310
[18:55:29.645] stderr> OpenSSH_8.1p1, LibreSSL 2.7.3
[18:55:30.337] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:urIdr3QLtY+DsFJgd4lyL+l8IN3pDKZFn5DDZaAQ0rg
[18:55:30.591] stderr> Authenticated to 192.168.0.12 ([192.168.0.12]:22).
[18:55:30.708] > Windows PowerShell 
> Copyright (C) 2009 Microsoft Corporation. All rights reserved.
[18:55:46.438] Terminating local server
[18:55:46.441] Resolver error: Error: Connecting with SSH timed out
	at g.Timeout (/Users/schmidt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.90.1/out/extension.js:1:585348)
	at Timeout._onTimeout (/Users/schmidt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.90.1/out/extension.js:1:630527)
	at listOnTimeout (node:internal/timers:559:17)
	at process.processTimers (node:internal/timers:502:7)
[18:55:46.446] Local server exit: 15
[18:55:46.467] ------
```
Log with `"remote.SSH.useLocalServer": false`

```
[18:34:49.159] Log Level: 2
[18:34:49.160] remote-ssh@0.90.1
[18:34:49.161] darwin x64
[18:34:49.168] SSH Resolver called for "ssh-remote+192.168.0.11", attempt 1
[18:34:49.169] "remote.SSH.useLocalServer": false
[18:34:49.170] "remote.SSH.showLoginTerminal": true
[18:34:49.171] "remote.SSH.remotePlatform": {"192.168.0.11":"windows","192.168.0.10":"linux","raspberrypi.local":"linux","192.168.0.9":"linux","192.168.0.12":"windows"}
[18:34:49.171] "remote.SSH.path": undefined
[18:34:49.171] "remote.SSH.configFile": undefined
[18:34:49.172] "remote.SSH.useFlock": true
[18:34:49.172] "remote.SSH.lockfilesInTmp": false
[18:34:49.172] "remote.SSH.localServerDownload": auto
[18:34:49.172] "remote.SSH.remoteServerListenOnSocket": false
[18:34:49.173] "remote.SSH.showLoginTerminal": true
[18:34:49.173] "remote.SSH.defaultExtensions": []
[18:34:49.173] "remote.SSH.loglevel": 2
[18:34:49.174] "remote.SSH.maxReconnectionAttempts": 0
[18:34:49.174] "remote.SSH.enableDynamicForwarding": true
[18:34:49.175] "remote.SSH.enableRemoteCommand": false
[18:34:49.175] "remote.SSH.serverPickPortsFromRange": {}
[18:34:49.176] "remote.SSH.serverInstallPath": {}
[18:34:49.213] SSH Resolver called for host: 192.168.0.11
[18:34:49.213] Setting up SSH remote "192.168.0.11"
[18:34:49.226] Using commit id "d045a5eda657f4d7b676dedbfa7aab8207f8a075" and quality "stable" for server
[18:34:49.235] Install and start server if needed
[18:34:49.265] PATH: /opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/schmidt/.rvm/bin
[18:34:49.266] Checking ssh with "ssh -V"
[18:34:49.300] > OpenSSH_8.1p1, LibreSSL 2.7.3

[18:34:49.305] Remote command length: 6205/8192 characters
[18:34:49.305] Running script with connection command: ssh -T -D 55085 -o ConnectTimeout=15 '192.168.0.11' powershell
[18:35:06.309] Resolver error: Error: Connecting with SSH timed out
	at g.Timeout (/Users/schmidt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.90.1/out/extension.js:1:585348)
	at Timeout._onTimeout (/Users/schmidt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.90.1/out/extension.js:1:679743)
	at listOnTimeout (node:internal/timers:559:17)
	at process.processTimers (node:internal/timers:502:7)
[18:35:06.317] ------

```

I can ssh successfully outside of VS Code via a terminal which defaults to cmd.exe:

```
ssh 192.168.0.11

Microsoft Windows [Version 6.1.7601]                                            
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.                 

jeff@WINDOWS-7-VM C:\Users\jeff>echo "this works"                               
"this works"       
```                                                             

When I try to run the precise VS Code command including powershell, it appears to connect but the terminal is not interactive (I'm not sure if this is normal or not):

```
$ ssh -T -D 55085 -o ConnectTimeout=15 '192.168.0.11' powershell
Windows PowerShell 
Copyright (C) 2009 Microsoft Corporation. All rights reserved.

echo 'foo'
this input is not interactive
etc
etc
test
dir

```

**Comment by eleanorjboyd** [comment]
Hello! Could you try adding the echo command in the same command as when you connect to your remote? For example using something like this `echo "echo hello" | ssh ...`

**Comment by jeff94040** [comment]
I was able to resolve the issue only by upgrading the remote OS to Windows 10. I had no luck with Windows 7 or Windows 8.1, seemingly no matter what I tried. The command `ssh -T -D 55085 -o ConnectTimeout=15 '192.168.0.xxx' powershell`, when run directly from the host command line, would connect but NOT present an interactive shell if the remote OS was Win7 or Win8.1, but would present an interactive shell only with Win10. I think this was the culprit and have since deleted the Win7/8.1 VMs so am no longer in a position to troubleshoot further.
